File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
    let max = 0;
    let ss = "";

    for (let i = 0; i < s.length; i++) {
        if (ss.includes(s[i] + "")) {
            max = Math.max(max, ss.length);
            ss = ss.substring(ss.indexOf(s.charAt(i)) + 1);
        }
        ss += s[i];
    }

    return Math.max(max, ss.length);
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max = 0;
        String ss = "";

        for (int i = 0; i < s.length(); i++) {
            if (ss.contains(s.charAt(i) + "")) {
                max = Math.max(max, ss.length());
                ss = ss.substring(ss.indexOf(s.charAt(i)) + 1); 
            }
            ss += s.charAt(i);
        }

        return Math.max(max, ss.length());
    }
}
